### PR TITLE
Support multiple nodes on DOM4 API methods

### PR DESCRIPTION
**Pull request by jonathantneal**
[DOM4](https://dom.spec.whatwg.org/) mutation methods accept multiple elements. Feature parity would be helpful between DOM4 and postcss.

- Supports multiple nodes on [append](https://dom.spec.whatwg.org/#dom-parentnode-appendnodes), [prepend](https://dom.spec.whatwg.org/#dom-parentnode-prependnodes), and [replaceWith](https://dom.spec.whatwg.org/#dom-childnode-replacewithnodes).
- Adds tests.
- Updates API documentation.

### CODE REVIEW

**Comment by ai** [comment]
I can’t accept it right now. Only as part of 5.0 API changes.

**Comment by ai** [comment]
Or maybe as part of 4.2, because it will not broke a API.

**Comment by ai** [comment]
Should we remove in 5.0 also arrays? `append(array)` to use only this method?

File: docs/api.md [hunk 1]
```diff
@@ -1036,9 +1036,9 @@ if ( decl.prop.match(/^-webkit-/) ) {
 }
 ```
 
-### `node.replaceWith(otherNode)`
+### `node.replaceWith(otherNodes...)`
```
Comment (lydell) [inline docs/api.md]: The dots should be on the other side. This is JavaScript, not CoffeeScript.

File: lib/at-rule.js [hunk 2]
```diff
@@ -27,14 +27,14 @@ export default class AtRule extends Container {
         }
     }
 
-    append(child) {
+    append() {
```
Comment (lydell) [inline lib/at-rule.js]: Why not use `...children`? Performance?

File: lib/at-rule.js [hunk 3]
```diff
@@ -27,14 +27,14 @@ export default class AtRule extends Container {
...
         if ( !this.nodes ) this.nodes = [];
-        return super.append(child);
+        return super.append.apply(this, arguments);
```
Comment (lydell) [inline lib/at-rule.js]: Same thing here. Why not `super.append(...children)`?

File: lib/container.js [hunk 4]
```diff
@@ -150,18 +150,35 @@ export default class Container extends Node {
         });
     }
 
-    append(child) {
-        let nodes = this.normalize(child, this.last);
-        for ( let node of nodes ) this.nodes.push(node);
+    append() {
+        for (let index in arguments) {
```
Comment (lydell) [inline lib/container.js]: A `for-in` to loop over an array-like object? Really?

**Comment by jonathantneal** [comment]
@lydell, thank you for showing me all of these wonderful ES6 goodies. If you cannot yet tell, I have spent a long time in ES5.

If you have time, hop in the [postcss gitter room](https://gitter.im/postcss/postcss). We can work together to improve this pull for 4.2.

File: lib/at-rule.js (hunk 3 above)
Comment (MoOx) [inline lib/at-rule.js]: Maybe because @jonathantneal is not aware of this es feature (babel use used and add support for es spread operator)

**Comment by MoOx** [comment]
Looks great !
+1 for this PR in the next minor release

**Comment by ai** [comment]
@lydell ES6 syntax is too new for us. I just know some stuff just today from @gaearon.

@jonathantneal wait for PR updates with more ES6 :). Welcome aboard :).

**Comment by MoOx** [comment]
That might helps you http://babeljs.io/docs/learn-es2015/

**Comment by lydell** [comment]
I shouldn’t review code when I’m too tired … that’s not the way I intended to sound! :/

**Comment by ai** [comment]
@jonathantneal I am ready to merge it with 4.2 branch. Just wait for ES6 fixes :).

**Comment by ai** [comment]
Merged

**Pull request closed without merging**